### PR TITLE
Slight refactor of part of e2e decommission test

### PR DESCRIPTION
**Pull request by davidwding**
As suggested in https://github.com/cockroachdb/cockroach-operator/issues/631, the PVC check within the decommission e2e test has been slightly refactored to untangle fetching the PVCs and running the test logic itself.

The important part of this PR is in the `RequireNumberOfPVCs` func in require.go; the rest is minor spelling fixes I encountered while reading through / running the code, or auto-formatting applied by crlfmt.

This PR does not also refactor `RequirePVCToResize` within the pvcresize e2e test because I haven't figured out how to actually run that test yet.

### CODE REVIEW

**Comment by chrislovecnm** [comment]
@davidwding RequirePVCToResize has to run on GKE, since kind and other platforms do not support resizing PVCs.  Take a look at the e2e test for gke in the Makefile

**Review by chrislovecnm — CHANGES_REQUESTED** [review]
Mostly nit picks.  I appreciate you doing spelling fixes as well, but next time can we please do them in a different PR, that way we ensure that the PR does not change code funcationality.

File: pkg/testutil/require.go [hunk 1]
```diff
@@ -46,11 +44,14 @@ import (
 	apiresource "k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
 )
 
 // RequireClusterToBeReadyEventuallyTimeout tests to see if a statefulset has started correctly and
 // all of the pods are running.
-func RequireClusterToBeReadyEventuallyTimeout(t *testing.T, sb testenv.DiffingSandbox, b ClusterBuilder, timeout time.Duration) {
+func RequireClusterToBeReadyEventuallyTimeout(
```
[the remark this answers is not in the document]
Comment (davidwding) [reply pkg/testutil/require.go:52]: I think this was done by our internal formatter `crlfmt`, which I was told to have my code editor auto-run every time I save. It seems like it enforces style beyond what `gofmt` recommends. Not sure if you have access to this but there's more info here: https://cockroachlabs.atlassian.net/wiki/spaces/CRDB/pages/73072641/What+is+a+Good+CockroachDB+PR#WhatisaGoodCockroachDBPR-Adherencetothecodingstyle

File: pkg/testutil/require.go [hunk 2]
```diff
@@ -615,40 +641,47 @@ func getPodLog(ctx context.Context, podName string, namespace string, clientset
 	return buf.String(), nil
 }
 
-// RequirePVCToResize checks that the PVCs are resized correctly
-func RequireNumberOfPVCs(t *testing.T, ctx context.Context, sb testenv.DiffingSandbox, b ClusterBuilder, quantity int) {
+// RequireNumberOfPVCs checks that the number of PVCs is as specified
+func RequireNumberOfPVCs(
```
[the remark this answers is not in the document]
Comment (davidwding) [reply pkg/testutil/require.go:645]: Can you elaborate? It's also iterating over the list of PVCs and checking that the quantity matches the input argument, so it still seems right to me.

File: go.sum [hunk 3]
```diff
@@ -1,5 +1,4 @@
 bazil.org/fuse v0.0.0-20160811212531-371fbbdaa898/go.mod h1:Xbm+BRKSBEpa4q4hTSxohYNQpsxXPbPry4JJWOB3LB8=
-bazil.org/fuse v0.0.0-20180421153158-65cc252bf669 h1:FNCRpXiquG1aoyqcIWVFmpTSKVcx2bQD38uZZeGtdlw=
```
[the remark this answers is not in the document]
Comment (davidwding) [reply go.sum]: Sure

**Review by davidwding — COMMENTED** [review]
Noted about keeping spelling/style changes in separate PRs for more cohesion within one PR; will be more diligent about that

File: pkg/testutil/require.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (chrislovecnm) [reply pkg/testutil/require.go:52]: Cool! But that is for crdb, not the operator. We have never used the tool on this project. @udnay should we use this tool now?

File: pkg/testutil/require.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (chrislovecnm) [reply pkg/testutil/require.go:645]: I might not be reading the diff correctly. I will take a look in a bit

File: pkg/testutil/require.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (udnay) [reply pkg/testutil/require.go:52]: Let's not use it for now @davidwding and we can discuss as a team. A lot of the docs are CRDB specific and I don't think that we need to adhere to all the same standards for now.

[the remark this answers is not in the document]
Comment (davidwding) [reply pkg/testutil/require.go:52]: SGTM

**Review by pseudomuto — DISMISSED** [review]

**Review by pseudomuto — APPROVED** [review]

**Pull request merged**